exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -74,7 +75,17 @@ func (q *boundedMemoryQueue) Produce(item Request) bool {
 	// their combined size is stored in q.size, and their combined capacity
 	// should match the capacity of the new queue
 	if q.size.Load() >= q.capacity {
-		return false
+		if q.overflow == nil {
+			return false
+		}
+		select {
+		case old := <-q.items:
+			q.size.Add(^uint32(0))
+			q.overflow(old)
+		default:
+			q.overflow(item)
```

Comment:
The docstring for this method says it returns false if the queue overflows, that's no longer the case here. The docs should be updated